### ISSUE TITLE
`compressPublicAssets` keeps uncompressed files

### ISSUE THREAD

**Issue by hybridherbst** [nitrojs/nitro#2702]
### Describe the feature

I'm not sure if this is intentional, but it causes issues on some servers without strict order definitions for compressed and uncompressed files.

Currently, when enabling `compressPublicAssets`, the originals are kept instead of deleted.

For reference, vite-compression has an option to delete the originals:
https://github.com/vbenjs/vite-plugin-compression?tab=readme-ov-file#options

Is there a way to delete the originals when `compressPublicAssets` is on?

### Additional information

- [ ] Would you be willing to help implement this feature?

**Comment by paulcrussell** [comment]
I don't think this is a valid issue. The server has to always be capable of returning the valid response _for the client_. In some cases a client can only handle an uncompressed document and that is ok and completely normal request. In this case the server should return the uncompressed document. 
This is known as Content Negotiation and is described at [Content Negotiation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Guides/Content_negotiation)

**Comment by hybridherbst** [comment]
That's true but it would be the server's task to serve that file – when there's only gzipped assets a server will unpack those before serving to the best of my understanding. It's not typical to have two files (one gzipped and one not) on a server.

**Comment by paulcrussell** [comment]
In your suggested approach, uncompressing is an expensive operation that would then kill performance of the server response.

The current approach taken is correct IMO as it keeps expensive operations during build time. I.e. compression. Having the compressed and uncompressed file available for the app process to serve the correct file is necessary for content negotiation to work.

What are you trying to optimize for? Storage use?: It's cheap and not worth sacrificing server response times.
